lib/modules.nix

```diff
@@ -87,6 +84,16 @@ rec {
       result = { inherit options config; };
     in result;
 
+
+ /* Filter disabled modules. Modules can be disabled allowing
+    their implementation to be replaced. */
+ filterModules = modules:
+   let
+     moduleKey = m: if isString m then toString ../nixos/modules + "/" + m else toString m;
+     disabledKeys = map moduleKey (concatMap (m: m.disabledModules) modules);
```

Comment:
Design question, before accepting this change.  Do we want to exclude a single module, or do we want to exclude a single module and all the modules that are inherited by it?